### ISSUE TITLE
Check failure when running tensorflow.python.ops.gen_array_ops.quantize_and_dequantize_v4

### ISSUE THREAD

**Issue by dmc1778** [tensorflow/tensorflow#59131]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.11.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Check failure when running .python.ops.gen_array_ops.quantize_and_dequantize_v4
```

### Standalone code to reproduce the issue

```
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_array_ops
try:
  arg_0_tensor = tf.saturate_cast(tf.random.uniform([2, 3, 4], minval=0, maxval=2, dtype=tf.int64), dtype=tf.uint64)
  arg_0 = tf.identity(arg_0_tensor)
  input_min = -1
  input_max = -1
  signed_input = False
  num_bits = 7
  range_given = True
  round_mode = "nan"
  narrow_range = True
  axis = -2
  out = gen_array_ops.quantize_and_dequantize_v4(arg_0,input_min=input_min,input_max=input_max,signed_input=signed_input,num_bits=num_bits,range_given=range_given,round_mode=round_mode,narrow_range=narrow_range,axis=axis,)
except Exception as e:
  print("Error:"+str(e))
```

### Relevant log output

```shell
023-01-06 09:14:31.245795: F ./tensorflow/python/eager/pywrap_tensor_conversion.h:58] Check failed: !PyErr_Occurred() 
Aborted

```
```
</details>

**Comment by mohantym** [comment]
Hi @nimashiri !

Thanks for sharing your gen_array_ops.quantize_and_dequantize_v4  w.r.to 2.11 version and Ubuntu.

I am getting below error with TF 2.11 ,  Ubuntu 18 and Cuda 11.2 + CudNN 8.1
```

Error:Value for attr 'T' of uint64 is not in the list of allowed values: bfloat16, half, float, double
	; NodeDef: {{node QuantizeAndDequantizeV4}}; Op<name=QuantizeAndDequantizeV4; signature=input:T, input_min:T, input_max:T -> output:T; attr=signed_input:bool,default=true; attr=num_bits:int,default=8; attr=range_given:bool,default=false; attr=T:type,allowed=[DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE]; attr=round_mode:string,default="HALF_TO_EVEN",allowed=["HALF_TO_EVEN", "HALF_UP"]; attr=narrow_range:bool,default=false; attr=axis:int,default=-1> [Op:QuantizeAndDequantizeV4]
```

Could you check with Cuda 11.2 , CudNN 8.1 and TF 2.11  or Cuda 11.7 , CudNN 8.2 and TF-nightly and let us know.

It would be great if you can the share OS and Cuda Details too. 
(Ubuntu Arm 64 or X86_64 , Cuda 11.2 or 11.7 )

Thank you!

**Comment by nimashiri** [comment]
From colab:

```
Error:Value for attr 'T' of uint64 is not in the list of allowed values: bfloat16, half, float, double
	; NodeDef: {{node QuantizeAndDequantizeV4}}; Op<name=QuantizeAndDequantizeV4; signature=input:T, input_min:T, input_max:T -> output:T; attr=signed_input:bool,default=true; attr=num_bits:int,default=8; attr=range_given:bool,default=false; attr=T:type,allowed=[DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE]; attr=round_mode:string,default="HALF_TO_EVEN",allowed=["HALF_TO_EVEN", "HALF_UP"]; attr=narrow_range:bool,default=false; attr=axis:int,default=-1> [Op:QuantizeAndDequantizeV4]

Colab paid products - Cancel contracts here

```

From terminal:

```
2023-01-09 10:11:50.975850: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-09 10:11:51.507938: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/home/nimashiri/anaconda3/envs/cuda11.2/lib/
2023-01-09 10:11:51.508097: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/home/nimashiri/anaconda3/envs/cuda11.2/lib/
2023-01-09 10:11:51.508104: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-09 10:11:51.997861: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.023990: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.024113: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.024507: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-09 10:11:52.025015: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.025121: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.025214: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.639793: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.639941: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.640069: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:11:52.640201: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1613] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 4125 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1660 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5
Error:can't convert negative int to unsigned
```

Simply running the script where the above code resides through VSCODE:

```
2023-01-09 10:13:02.292944: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-09 10:13:02.852646: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory
2023-01-09 10:13:02.852689: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory
2023-01-09 10:13:02.852696: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-09 10:13:03.378474: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-09 10:13:03.415108: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory
2023-01-09 10:13:03.415132: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-09 10:13:03.415379: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Error:can't convert negative int to unsigned
```

Running via ```subprocess.run()```:

```
2023-01-09 10:15:16.155685: F ./tensorflow/python/eager/pywrap_tensor_conversion.h:58] Check failed: !PyErr_Occurred() 
Aborted

```

**Comment by mihaimaruseac** [comment]
@learning-to-play

**Comment by Gopi-Uppari** [comment]
Hi @dmc1778,

I reproduced the error using the code provided above in Google Colab. The issue arises because the `QuantizeAndDequantizeV4` operation only supports specific floating-point data types for its input tensor (`arg_0`) and attributes (`input_min`, `input_max`). It does not support `uint64`, which is being used in the code.

To resolve this, please use `tf.float32` for the input tensor and attributes, then verify that it works. For reference, you can check this [gist file](https://colab.research.google.com/gist/Gopi-Uppari/b1cf10ad67d154d389bea840a6cbffe5/tf_issue_59131.ipynb).

Thank you.
